### PR TITLE
Add redux to my react project

### PR DESCRIPTION
**Pull request by mumbereh**
### **In this milestone I implemented the following**
Add Redux Toolkit (npm install react-redux @reduxjs/toolkit).
Create a directory that will contain all  Redux logic (/src/redux)
Configure a Redux store (/src/redux/store.js)
Define a slice of state for books that:
Includes an array of books (initial state: empty array)
Includes a reducer that adds a book
Includes a reducer that removes a book
Define a slice of state for categories that:
Includes an array of categories (initial state: empty array)

### CODE REVIEW

File: src/redux/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      state.books.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book.id !== action.payload);
+    },
+  },
+});
+
+export const { addBook, removeBook } = booksSlice.actions;
+export default booksSlice.reducer;
```
Comment (thecodechaser) [inline src/redux/booksSlice.js]: - [ ] Please make sure to place the booksslice in the books folder to follow the duck pattern as requested in the project requirements. example: `src/redux/books/booksSlice.js`

image

File: src/redux/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  status: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
+});
+
+export default categoriesSlice.reducer;
```
Comment (thecodechaser) [inline src/redux/categoriesSlice.js]: - [ ] Please make sure to place the categoriesSlice in the categories folder to follow the duck pattern as requested in the project requirements. example: `src/redux/categories/categoriesSlice.js`

image

File: src/redux/categoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+const initialState = {
+  status: 'Under construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {},
+});
```
Comment (thecodechaser) [inline src/redux/categoriesSlice.js]: - [ ] Please make sure to add the checkstatus reducer as requested in the project requirements.
   - You can consider using the below code.
 
```ruby
   reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @mumbereh👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Good job with redux code. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @mumbereh, 👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... It's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descrtipive PR⭐
- [ ] Kudos for working on the functionality of the redux🤩
- [ ] No Linters error ✅ 
- [ ] Follow DUCK pattern 💯🎉


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**